assignment-client/src/assets/AssetServer.cpp

```diff
@@ -0,0 +1,279 @@
+//
+//  AssetServer.cpp
+//
+//  Created by Ryan Huffman on 2015/07/21
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+
+#include "AssetServer.h"
+
+#include <QBuffer>
+#include <QCryptographicHash>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QCoreApplication>
+#include <QEventLoop>
+#include <QRunnable>
+#include <QString>
+
+#include <NodeType.h>
+
+const QString ASSET_SERVER_LOGGING_TARGET_NAME = "asset-server";
+
+void writeError(NLPacketList* packetList, AssetServerError error) {
+    packetList->writePrimitive(error);
+}
+
+class SendAssetTask : public QRunnable {
+public:
+    SendAssetTask(MessageID messageID, const QByteArray& assetHash, QString filePath, DataOffset start, DataOffset end, const SharedNodePointer& sendToNode) :
+        QRunnable(),
+        _messageID(messageID),
+        _assetHash(assetHash),
+        _filePath(filePath),
+        _start(start),
+        _end(end),
+        _sendToNode(sendToNode)
+    {
+    }
```

Comment:
I think by the coding standard with this many member variable initializations we'll need this to be in the cpp